internal/statestore/backfill_test.go

```diff
@@ -0,0 +1,303 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package statestore
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/gomodule/redigo/redis"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	utilTesting "open-match.dev/open-match/internal/util/testing"
+	"open-match.dev/open-match/pkg/pb"
+)
+
+func TestCreateBackfill(t *testing.T) {
+	cfg, closer := createRedis(t, false, "")
+	defer closer()
+	service := New(cfg)
+	require.NotNil(t, service)
+	defer service.Close()
+	ctx := utilTesting.NewContext(t)
+
+	bf := pb.Backfill{
+		Id:         "1",
+		Generation: 1,
+	}
+
+	var testCases = []struct {
+		description     string
+		backfill        *pb.Backfill
+		ticketIDs       []string
+		expectedCode    codes.Code
+		expectedMessage string
+	}{
+		{
+			description:     "ok, backfill is passed, ticketIDs is nil",
+			backfill:        &bf,
+			ticketIDs:       []string{"1", "2"},
+			expectedCode:    codes.OK,
+			expectedMessage: "",
+		},
+		{
+			description:     "create existing backfill, err expected",
+			backfill:        &bf,
+			ticketIDs:       nil,
+			expectedCode:    codes.AlreadyExists,
+			expectedMessage: "backfill already exists, id: 1",
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.description, func(t *testing.T) {
+			err := service.CreateBackfill(ctx, tc.backfill, tc.ticketIDs)
+			if tc.expectedCode == codes.OK {
+				require.NoError(t, err)
+			} else {
+				require.Error(t, err)
+				require.Equal(t, tc.expectedCode.String(), status.Convert(err).Code().String())
+				require.Contains(t, status.Convert(err).Message(), tc.expectedMessage)
+			}
+		})
+	}
+
+	// pass an expired context, err expected
```

Comment:
This is a different test, it belongs in its own test function.

Here,  and for the other backfill functions.

Reply:
@Laremere  Why? I use high-level names for tests where I include multiple test cases. 
For instance -TestCreateBackfill means that I test CreateBackfill under different conditions, and passing an expired context can be considered as another test case. 

Exactly the same happens here:
https://github.com/googleforgames/open-match/blob/master/internal/app/frontend/frontend_service_test.go#L45

Expire context scenario is included in a general TestDoCreateTickets function. In my case, I didn't add preAction, but tested expired context case at the end of the function.

If you want to extract expired context test case to its own function it should be done in all tests, not only in backfill_test.go, which can take some time I guess. What do you think?